src/compiler/frontend/expander/define.rs

```diff
@@ -46,7 +48,7 @@ impl Expander {
                                 exprs,
                                 datum.source_location().clone()
                             );
-                            let expanded_lambda = self.expand_macros(&lambda)?;
+                            let expanded_lambda = self.expand_macros(&lambda)?.expect("expected lambda body to expand correctly");
```

Comment:
When the parse succeeds without producing an expression, this indicates a compiler bug. Surface that consistently instead of using `expect`